### PR TITLE
Enable correct loading of settings from environment variables

### LINKED ISSUE #2370: SettingsManager overwriting env var settings
## 🐛 Bug
Pydantic BaseSettings provides a nice way to set settings using [environment variables](https://pydantic-docs.helpmanual.io/usage/settings/#environment-variable-names), but at the moment, the `SettingsManager._load` method is overwriting this:

```python
# test.py
from napari.utils.settings import SETTINGS, CORE_SETTINGS

print("Application Settings:", CORE_SETTINGS[0]().theme)
print("SettingsManager:", SETTINGS.application.theme)
```

```sh
$ NAPARI_THEME=light python test.py
Application Settings: light
SettingsManager: dark
```

In the case where a settings is defined multiple ways, I think we should aim to match the [default field priority](https://pydantic-docs.helpmanual.io/usage/settings/#field-value-priority) in pydantic:

- Arguments passed to the `Settings` class initialiser.
- Environment variables, e.g. `napari_setting`
- Variables loaded from a dotenv (.env) file, (or, in our case, yaml settings) 
- Variables loaded from the secrets directory.
- The default field values for the Settings model.

This does open a bit of a challenge for the concept of persistence, since I think that loading with `NAPARI_THEME=light` would probably then override the yaml file itself on close... which is less desirable (I think it would be nice to be able to use environment variables to override settings at startup, without permanently changing the preferences)

### PR DESCRIPTION
**Pull request by goanpeca**
# Description

Fixes #2370

## Type of change

- [x] Bug-fix (non-breaking change which fixes an issue)

# How has this been tested?

- [x] example: all tests pass with my change

## Final checklist:
- [x] My PR is the minimum possible work for the desired functionality
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] If I included new strings, I have used `trans.` to make them localizable.
      For more information see our [translations guide](https://napari.org/docs/dev/guides/translations.html).

### CODE REVIEW

**Comment by goanpeca** [comment]
This now fixes loading settings from env vars. I have a question though on what should the behavior be when someone set an env variable with an incorrect value.

For example:

```bash
NAPARI_THEME=foobar napari
```

So, NAPARI_THEME should be either `"light"` or `"dark"` (without any theme plugins). 

Now:
1. We fail loudly and napari will not start (current behavior)
2. We warn and use the default or whatever is set in the settings 

The example of theme is a trivial one, where setting the wrong value and have a warning is fine. For other future settings it might be better to have a "loud" fali/exception. 

Failing loudly makes it clear and enforces using the right values from the get go but may be a bit annoying.

Warning might be ok for some settings but not for others and the actual wanr might be lost in the console.

I would like to hear opinions @napari/core-devs 🙃 

Thanks!

File: napari/utils/settings/_manager.py [hunk 1]
```diff
@@ -117,53 +110,43 @@ def _save(self):
...
-                            model_data.pop(item)
-
-                    model_data.update(model_data_replace)
-                    model = self._models[section](**model_data)
-                    model.events.connect(lambda x: self._save())
-                    self._settings[section] = model
+            # Load data once and save it in the base class
+            BaseNapariSettings._LOADED_DATA = data
+
+        for setting in CORE_SETTINGS:
```
Comment (tlambert03) [inline napari/utils/settings/_manager.py:131]: 👍  like this diff

File: napari/utils/settings/_defaults.py [hunk 2]
```diff
@@ -175,17 +175,70 @@ class QtBindingChoice(str, Enum):
...
+
+        # Get defaults from the schema
+        model_schema = model_class.schema()
+        section = model_schema["section"]
+        default_properties = model_schema.get("properties", {})
+        yaml_settings = loaded_data.get(section, {}).copy()
+
+        try:
+            model_class(**yaml_settings)
+        except KeyError:
```
Comment (tlambert03) [inline napari/utils/settings/_defaults.py]: what causes the potential KeyError happen here?

**Review by tlambert03 — APPROVED** [review]
i like this solution!  thanks @goanpeca 

tangential question: I assume there's a good reason, but I can't remember... can SettingsManager itself not be a `BaseSettings` with the sub-settings as fields?

**Comment by tlambert03** [comment]
> Failing loudly makes it clear and enforces using the right values from the get go but may be a bit annoying.
> Warning might be ok for some settings but not for others and the actual wanr might be lost in the console.
> I would like to hear opinions @napari/core-devs 🙃

I don't have a really strong opinion here.  As discussed on zulip, my inclination was to warn... but I'm totally fine with an error as well.

File: napari/utils/settings/_defaults.py (hunk 2 above)
Comment (goanpeca) [reply napari/utils/settings/_defaults.py]: It came from the original code I copied 
https://github.com/napari/napari/blob/master/napari/utils/settings/_manager.py#L147

Indeed it is now not needed. Good catch 🎸 !

**Comment by goanpeca** [comment]
>tangential question: I assume there's a good reason, but I can't remember... can SettingsManager itself not be a BaseSettings with the sub-settings as fields?

That is a good question 🤔 

To be honest I did not explore that option at all, so I will open an issue and look at that in a different PR. Thanks for raising it!

File: napari/utils/settings/_defaults.py (hunk 2 above)
Comment (goanpeca) [reply napari/utils/settings/_defaults.py]: Fixed!

**Review by sofroniewn — APPROVED** [review]
LGTM!

**Pull request merged**